### PR TITLE
Fixed all tests

### PR DESCRIPTION
**Pull request by HalidOdat**
Fixed all tests from the new parser #281
```
test result: ok. 127 passed; 0 failed; 1 ignored; 0 measured; 0 filtered out
```
:tada: :tada: :tada:

### CODE REVIEW

File: boa/src/exec/mod.rs [hunk 1]
```diff
@@ -538,7 +558,10 @@ impl Executor for Interpreter {
 
                 Ok(obj)
             }
-            _ => unimplemented!(),
+            Node::Spread(ref node) => {
+                Ok(Gc::new((*self.run(node)?).clone())) // for now we can do nothing but return the value as-is
```
Comment (Razican) [inline boa/src/exec/mod.rs]: I would add a `TODO` somewhere here to search for it easier.

File: boa/src/exec/mod.rs [hunk 2]
```diff
@@ -591,7 +614,7 @@ impl Interpreter {
                                     .environment
                                     .initialize_binding(name, expr.clone());
                             }
-                            Node::UnaryOp(UnaryOp::Spread, ref expr) => {
+                            Node::Spread(ref expr) => {
```
Comment (Razican) [inline boa/src/exec/mod.rs]: Why was this moved out of `UnaryOp`?

Comment (HalidOdat) [reply boa/src/exec/mod.rs]: In the standard/specification `Spread` is not a part of the unary expression.
see: https://tc39.es/ecma262/#prod-UnaryExpression

**Comment by jasonwilliams** [comment]
Wow, good work!
